synapse/api/errors.py

```diff
@@ -307,6 +312,32 @@ def __init__(
         super().__init__(code, msg, errcode, additional_fields)
 
 
+class UnstableSpecAuthError(AuthError):
```

Comment:
I've tried to be careful here by subclassing AuthError, in the hope that any internal logic expecting an AuthError to pop out won't behave any differently.